app/components/progress_bar_component.rb

```diff
@@ -1,11 +1,66 @@
-class ProgressBarComponent < ViewComponent::Base
-  def initialize(current_user, programme, title: nil, body: nil, steps_to_accreditation: true)
+class ProgressBarComponent < CmsWithAsidesComponent
+  delegate :current_user, to: :helpers
+
+  def initialize(programme:, aside_slug: nil, title: nil, body: nil)
+    aside_sections = if aside_slug.nil?
+      nil
+    else
+      [{slug: aside_slug}]
+    end
+
+    super(aside_sections:)
     @programme = programme
-    @current_user = current_user
     @title = title
     @body = body
-    @steps_to_accreditation = steps_to_accreditation
 
     @programme_objectives = programme.programme_objectives_displayed_in_progress_bar
   end
+
+  def before_render
+    @current_user = current_user
+  end
+
+  private
+
+  def content_spacing_class(class_name)
+    if @programme.show_extra_objectives_on_progress_bar?
+      class_name + "-primary"
+    else
+      class_name
+    end
+  end
+
+  def user_enrolled_class
+    if @programme.user_enrolled?(@current_user)
+      "icon-ticked-circle"
+    else
+      "icon-blank-circle"
+    end
+  end
+
+  def course_bookings_status_class(objective, state)
+    current_state = objective.course_credit_state(current_user, state)
+
+    case current_state
+    when "required_credits"
+      "icon-ticked-circle"
+    when "missing_credits"
+      "icon-pending-circle"
+    else
+      "icon-blank-circle"
+    end
+  end
+
+  def multi_stage_objectives
```

Comment:
Could these be returned from the programme activity group? so its returns false or the objectives, that way we have some flexibility for the future?